src/main/java/uk/nhs/hee/tis/trainee/credentials/dto/CredentialDataDto.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.credentials.dto;
+
+/**
+ * An interface representing a DTO containing credential data.
+ */
+public interface CredentialDataDto {
+
```

Comment:
Is the intention to abstract commonalities from the various credential types into this interface, once we have more than one credential type and can see some areas of shared data/concern?

Reply:
The thinking behind this is to be able to limit what can be passed to the `CredentialService` and `JwtService` while still leaving them being generic.
We could just as easily pass an `Object` to those services and it would work exactly the same, but then we rely on the javadoc to explain what we expect it to be sent instead of the method signature dictating

Reply:
Ok, that makes sense - neat.

Reply:
Based on your below comment, it may also be a good place to declare a `getExpiry` type interface method at a later date.